### DISCUSSION TITLE
Working on an IOS app

### DISCUSSION THREAD

**Discussion by Chewbaccalakis** [mealie-recipes/mealie#2441] [General]
Since one doesn't exist and I personally love Mealie, I figured i may as well use this as an opportunity to learn Swift and create an IOS app for Mealie.

I'm currently stumped on getting the API to respond with all the categories and tags, does anyone know if there's more current API documentation than I'm finding or had any success with doing so and want to share how?

In addition to that, I have a large chunk of the app done but if anyone who has experience with Swift, especially SwiftUI wants to take a look and help me fine tune the UI plus some backend stuff that would be greatly appreciated. I plan on making the repo for it public once I have the app most of the way there (and hopefully have gone back and commented more of my code, at least somewhat) as well as publishing it on the app store.

I'll also happily listen to things people would want to see in an app for Mealie or whatever other requests you may have, although I make no promises as to whether I'm personally capable of implementing them.

**Comment by michael-genson** [comment]
> does anyone know if there's more current API documentation than I'm finding or had any success with doing so and want to share how?

You can always find the most up-to-date docs for your version at `https://YOUR-URL.com/docs`. If you're using the omni image, the docs link is currently bugged, but you can go directly to the backend locally by typing `YOUR-LOCAL-IP:9000/docs`

**Reply by Chewbaccalakis** [reply]
Perfect, Thanks! That's exactly what I was looking for. I saw reference to the local docs but couldn't actually find what the path was anywhere

**Reply by michael-genson** [reply]
No problem! While I can't help with Swift/IOS, if you run into any more API issues let me know

**Comment by fivestones** [comment]
I'm excited to hear this and would love to be able to use an iOS app for mealie! @Chewbaccalakis are you keeping your code in github or somewhere other people can take a look?

**Reply by Chewbaccalakis** [reply]
I am, I believe the repo is private currently. I mainly need to go through it and comment better plus organize a few things and I'll make the repo public. I hopefully should be back at it soon. I was in the process of switching jobs so that was eating up most of my time. After thanksgiving I'll try to sit down and do it

**Reply by fivestones** [reply]
Everybody (including me) thinks they should wait until their code is organized and well commented before making it public. But no one else cares. Would we like it more if it was public, organized, and well commented? Sure. But if the question is between unorganized, uncommented, public code or no code at all, 100% of us would say we'll take the unorganized, uncommented, but public code.
Plus if you make it public, we can help you clean it up 😀

**Comment by genebean** [comment]
Curious if you made any progress

**Comment by austinbru** [comment]
I second that. Also curious and would love to have iOS app support

**Reply by pgale** [reply]
Me too 😁

**Comment by ddiguy** [comment]
Checking in to see if you've made progress on this.  I have an Apple Developer account so I could help you get it on the app store if you don't have one.  I'm loving this program, but need an iPhone app.

**Comment by Scot-Survivor** [comment]
Yet another comment asking if progress has been made?

If now the PWA could be improved by caching more. 

Most supermarkets I'm in act as oversized faraday cage & getting signal can be a gamble.
